release.sh

```diff
@@ -89,80 +89,13 @@ function make_tarball()
     cp *.tar.gz ..
 }
 
-function make_rpm()
-{
-    if [ "x$BUILD_RPMS" != "xtrue" ]; then
```

Comment:
`BUILD_RPMS` is an option in the release CI job. This would be a good opportunity to remove that option there too. We're not building RPMs through the release job anyway.